File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int leftPtr = 0, rightPtr = 0;
        int numsSize = nums.size();
        while(rightPtr < numsSize){
            if(nums[rightPtr] != val){
                nums[leftPtr] = nums[rightPtr];
                rightPtr++;
                leftPtr++;
            }else{
                rightPtr++;
            }
        }
        return leftPtr;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        ans = 0
        leftPtr, rightPtr = 0, len(heights) - 1
        while leftPtr < rightPtr:
            curVol = (rightPtr - leftPtr) * min(heights[leftPtr], heights[rightPtr])
            if curVol > ans:
                ans = curVol

            if heights[leftPtr] < heights[rightPtr]:
                leftPtr+=1
            else:
                rightPtr-=1
        return ans
            